pkg/hs/urbit-king/lib/Urbit/King/Main.hs

```diff
@@ -586,11 +588,28 @@ runShip (CLI.Run pierPath) opts daemon = do
   where
     runPier :: MVar () -> RIO PierEnv ()
     runPier mStart = do
+      injections <- loadInjections (CLI.oInjectEvents opts)
       tryPlayShip
         (CLI.oExit opts)
         (CLI.oFullReplay opts)
         (CLI.oDryFrom opts)
         mStart
+        injections
+
+    loadInjections :: [CLI.Injection] -> RIO PierEnv [Ev]
+    loadInjections injections = do
+      perInjection :: [[Ev]] <- forM injections $ \case
+          CLI.InjectOneEvent filePath -> do
+            logInfo $ display $ "boot: reading injected event from " ++
+              (pack filePath :: Text)
```

Comment:
isn't `pack` always `Text`?

Reply:
We need the cast here because filePath is a FilePath, like all the times in this file where we have this exact same cast on `pack`.

Reply:
Elliot, pack is a function String -> Text, and FilePath = String. There are no casts in Haskell in the sense that there are in C++.

Reply:
Feel free to try it locally. It does not compile, nor do any of the other instances where this occurs.